osu.Game/Database/BackgroundDataStoreProcessor.cs

```diff
@@ -436,6 +437,95 @@ private void upgradeScoreRanks()
             completeNotification(notification, processedCount, scoreIds.Count, failedCount);
         }
 
+        private record ModMultiplierChange(
+            string ModAcronym,
+            string RulesetName,
+            double OldMultiplier,
+            double NewMultiplier,
+            int ScoreVersion);
+
+        private static readonly ModMultiplierChange[] mod_multiplier_changes =
+        [
+            new ModMultiplierChange("CL", "taiko", 0.96, 1, 30000017),
+            new ModMultiplierChange("CL", "fruits", 0.96, 1, 30000017),
+            new ModMultiplierChange("CL", "mania", 0.96, 1, 30000017),
+        ];
+
+        private void upgradeModMultipliers()
+        {
+            Logger.Log("Performing mod multiplier upgrades...");
+
+            int latestMultiplierChange = mod_multiplier_changes.Max(change => change.ScoreVersion);
+
+            var scores = realmAccess
+                         .Run(r => r.All<ScoreInfo>()
+                                    .Where(score => score.TotalScoreVersion < latestMultiplierChange && !score.BackgroundReprocessingFailed)
+                                    .Detach())
+                         // must be done after materialisation, as realm doesn't support
+                         // filtering on nested property predicates or projection via `.Select()`
+                         .Where(s => s.Ruleset.IsLegacyRuleset())
+                         .ToList();
+
+            if (scores.Count == 0)
+                return;
+
+            var notification = showProgressNotification(scores.Count, "Adjusting mod multipliers for scores", "scores now use latest mod multipliers");
+
+            int processedCount = 0;
+            int failedCount = 0;
+
+            foreach (var score in scores)
+            {
+                if (notification?.State == ProgressNotificationState.Cancelled)
+                    break;
+
+                updateNotificationProgress(notification, processedCount, scores.Count);
+
+                sleepIfRequired();
+
+                double? newTotalScore = null;
+
+                try
+                {
+                    foreach (var multiplierChange in mod_multiplier_changes)
+                    {
+                        if (score.TotalScoreVersion >= multiplierChange.ScoreVersion
+                            || score.Ruleset.ShortName != multiplierChange.RulesetName
+                            || score.APIMods.All(m => m.Acronym != multiplierChange.ModAcronym))
+                        {
+                            continue;
+                        }
+
+                        newTotalScore ??= score.TotalScore;
+                        newTotalScore /= multiplierChange.OldMultiplier;
+                        newTotalScore *= multiplierChange.NewMultiplier;
```

Comment:
I'm not sure I fully understand, but don't we possibly lose precision by doing this here ?

Reply:
Alright, at least said precision loss should be deterministic and not cause real issues. Thanks for clarifying !

Reply:
> Alright, at least said precision loss should be deterministic and not cause real issues. Thanks for clarifying !

this is not important, because we're already loosing 10000x more precision by converting stable scores to lazer